packages/react-search-ui/src/__tests__/withSearch.test.js

```diff
@@ -122,33 +125,33 @@ describe("withSearch", () => {
     }
 
     it("will inject specified state", () => {
-      const element = setup(["searchTerm"]);
+      const element = setup(({ searchTerm }) => ({ searchTerm }));
 
       expect(element.text()).toEqual("a search term");
     });
 
     it("will inject a specified action", () => {
-      const element = setup(["setSearchTerm"]);
+      const element = setup(({ setSearchTerm }) => ({ setSearchTerm }));
 
       expect(element.text()).toEqual("setSearchTerm");
     });
 
     it("will not inject an  unspecified state", () => {
-      const element = setup(["resultSearchTerm"]);
+      const element = setup(({ resultSearchTerm }) => ({ resultSearchTerm }));
 
       expect(element.text()).toEqual("another search term");
     });
 
-    it("will inject everything if an empty array is passed", () => {
-      const element = setup([]);
+    it("will inject everything if nothing is passed", () => {
```

Comment:
I don't have a strong opinion on this, so if you disagree, I'll back up.

I think that it would be better to follow the "performance by default" principle here, i.e. if you don't pass a list of props you're using, the component won't update.
The "poor performance" way could be opt-in with passing the entire state deliberately `withSearch(context => context)(Component)`.

I think if we go that way, the search-ui could avoid getting a reputation as a laggy library from those who try to setup a library as fast as possible and didn't read the docs carefully.

Also, if nothing is passed, we could give a console warn like "please provide a list of props".

Reply:
@yakhinvadim That's a fair point. It's also nice as we talked about before to have just 1 way to do things. One less decision to make, you just always pass a `mapContextToProps` function.

Reply:
One less thing to explain.

Reply:
@yakhinvadim https://github.com/elastic/search-ui/pull/249/commits/3d2b933700e1106ccc5fff6ec49d4cc213fde30f

Reply:
Great!